src/App.tsx

```diff
@@ -1,24 +1,196 @@
+        return todo.completed;
+      default:
+        return true;
+    }
+  }), [selectedStatus, todos]);
+
+  const countOfNotCompletedTodo = useMemo(() => todos
+    .filter((todo) => !todo.completed).length, [todos]);
+
+  const allTodoCompleted = useMemo(() => filteredTodo
+    .every(({ completed }) => completed), [filteredTodo]);
+
+  const atleastOneTodoCompleted = useMemo(() => filteredTodo
+    .some(({ completed }) => completed), [filteredTodo]);
+
+  const handleDeleteCompleted = async () => {
+    setIsLoading(true);
+
+    const requests = todos
+      .filter((todo) => todo.completed)
+      .map(async (todo) => {
+        setDeletedTodoIds(prevTodos => [...prevTodos, todo.id]);
+
+        await deleteTodo(todo.id);
+        setTodos(prev => prev.filter(({ id }) => todo.id !== id));
+      });
+
+    try {
+      await Promise.all(requests);
+
+      setIsLoading(false);
+      setErrorMessage('');
+
+      setTodos(prev => prev.filter(todo => !todo.completed));
+    } catch (errorUpdate) {
+      setIsLoading(false);
+
+      setErrorMessage('Unable to delete a todo');
+
+      setDeletedTodoIds(() => []);
+      // eslint-disable-next-line no-console
+      console.warn(errorUpdate);
+
+      hasDeleteTodoErrorTimerId.current = waitToClose(3000, setErrorMessage);
+    }
+  };
+
+  const handleToggleButton = async () => {
+    setIsLoading(true);
+
+    const requests = todos
+      .map(async (todo) => {
+        setDeletedTodoIds(prevTodos => [...prevTodos, todo.id]);
+
+        await updateTodo(todo.id, {
+          ...todo,
+          completed: !allTodoCompleted,
+        });
+      });
+
+    try {
+      await Promise.all(requests);
+
+      setTodos(prevTodos => prevTodos.map(prevTodo => ({
+        ...prevTodo,
+        completed: !allTodoCompleted,
+      })));
+
+      setDeletedTodoIds(() => []);
+
+      setErrorMessage('');
+      setIsLoading(false);
+    } catch (error) {
+      setErrorMessage('Unable to update a todo');
+
+      setIsLoading(false);
+      setDeletedTodoIds(() => []);
+    }
+  };
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <SearchField
+          searchValue={searchValue}
+          setSearchValue={setSearchValue}
+          allTodoCompleted={allTodoCompleted}
+          inputFieldRef={inputFieldRef}
+          handleToggleButton={handleToggleButton}
+          hasTodos={hasTodos}
+          setTodos={setTodos}
+          setTempTodo={setTempTodo}
+          hasAddTodoErrorTimerId={hasAddTodoErrorTimerId}
+        />
```

Comment:
Why do you call it `SearchField` if actually it doesn't search anything, but adds a new todo 🤯